### PR TITLE
Implement Personnel resource (front + back-end)

### PR DESCRIPTION
**Pull request by pseigo**
The data model and database logic lives in `backend/src/model`. The templates are in `backend/src/views`, as before. The glue between a client's request and business logic lives in `backend/src/routes`.

Most of the weight here is in `backend/src/model/accounts.ts` and `backend/src/routes/personnel/personnelRoute.ts`.

- The `Accounts` module has functions for all the CRUD operations on a Personnel. These functions make queries directly to the database.
- The `PersonnelRoute` takes user requests, extracts parameters, and asks the `Accounts` module to do the appropriate action. Each `Accounts` function returns a JS promise, so when the DB calls finish, the router renders the page and sends it back to the user.

Let me know if you have any questions! Go ahead and play with it a bit.

_p.s., sorry for all the promise verbosity, I don't know how to use await 🙃_

## Some notes

- Made client in Database class public so other classes can use it to make queries (we can also use a "Pool" if we want! Looks like it has the same API but probably sends queries off in batches)
- Removed redundant fields from Database (can re-add if needed)
- Created Accounts context with a Personnel type
- Only accounts.ts is responsible for actually making database calls. Anything that needs to hit the database for Personnel goes through the Accounts class

## Files added/changed

```diff
 README.md                                          |  15 +-
 backend/.env                                       |   5 +
 backend/.env.example                               |   6 +-
 backend/Dockerfile                                 |  19 ++
 backend/package-lock.json                          | 106 +++++++++++
 backend/package.json                               |   2 +
 backend/src/database.ts                            |  25 ++-
 backend/src/model/accounts.ts                      | 309 +++++++++++++++++++++++++++++++++
 backend/src/model/accounts/permissions.ts          |  15 ++
 backend/src/model/accounts/personnel.ts            |  47 +++++
 backend/src/public/scripts/app.js                  |  74 ++++++++
 backend/src/public/scripts/new_personnel.js        |  19 ++
 backend/src/public/stylesheets/base.css            |  16 ++
 backend/src/routes/mainRouter.ts                   |   4 +
 backend/src/routes/personnel/personnelRoute.ts     | 175 ++++++++++++++++++-
 backend/src/server.ts                              |   5 +-
 backend/src/views/backend/layout/foot.ejs          |   0
 backend/src/views/backend/layout/head.ejs          |   1 +
 backend/src/views/backend/layout/header.ejs        |   7 +
 backend/src/views/backend/page/about.ejs           |  23 +++
 backend/src/views/backend/page/index.ejs           |  54 +-----
 backend/src/views/backend/page/personnel/edit.ejs  |  80 +++++++++
 backend/src/views/backend/page/personnel/index.ejs |  63 +++++++
 backend/src/views/backend/page/personnel/new.ejs   |  75 ++++++++
 backend/src/views/backend/page/personnel/show.ejs  |  59 +++++++
 docker-compose.yml                                 |  12 ++
 26 files changed, 1152 insertions(+), 64 deletions(-)
```

### CODE REVIEW

File: backend/Dockerfile [hunk 1]
```diff
@@ -8,10 +8,12 @@ COPY package-lock.json ./
 
 COPY tsconfig.json ./
 
+copy build.ts ./
```
Comment (nuxeric) [inline backend/Dockerfile:11]: I think typo here unless Dockerfile lowercase is okay, if you've tested this its probably ifne

**Review by nuxeric — APPROVED** [review]
I see how you set up the project / where you're going with this. Awesome thanks for your hard work it looks like a great template for me and Shaun to go off of

File: backend/src/views/backend/page/index.ejs [hunk 2]
```diff
@@ -4,59 +4,17 @@
...
-          <div class="line-worker-responsibilities-container">
-            <label for="responsibilities">
-              <span>Responsibilites</span>
-            </label>
-            <textarea wrap="soft" id="responsibilities" name="line_worker_responsibilities" placeholder="Responsible for..."></textarea>
-          </div>
-          <button type="submit">Create user</button>
-          <button type="reset">Reset</button>
-        </form>
+        <a class="button" href="/personnel">Manage Personnel</a>
```
Comment (ShaunDonn) [inline backend/src/views/backend/page/index.ejs:14]: To confirm, this is where we would put all the buttons for the different functions, and each would take us to the corresponding page?

File: backend/src/views/backend/page/personnel/index.ejs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    <%- include('../../layout/head.ejs') -%>
+  </head>
+  <body>
+    <%- include('../../layout/header.ejs') -%>
+
+    <main>
+      <div class="hero">
+        <h1><%= title %></h1>
+
+        <a class="button" href="/personnel/new">New Personnel</a>
```
Comment (ShaunDonn) [inline backend/src/views/backend/page/personnel/index.ejs:13]: When this button is pressed, it prompts the screen/layout to allow the user to create a new personnel?

Comment (pseigo) [reply backend/src/views/backend/page/personnel/index.ejs:13]: Yep, just redirects to a different page

**Review by ShaunDonn — APPROVED** [review]
Looks really good to me!

File: backend/src/views/backend/page/personnel/index.ejs (hunk 3 above)
Comment (pseigo) [reply backend/src/views/backend/page/personnel/index.ejs:13]: There isn't any fancy JS stuff going on here, every screen is its own page :)

File: backend/src/views/backend/page/index.ejs (hunk 2 above)
Comment (pseigo) [reply backend/src/views/backend/page/index.ejs:14]: Yep yep

**Pull request merged**